tests/preflight/preflight_test.go

```diff
@@ -427,6 +436,394 @@ var _ = Describe("Preflight Tests", func() {
+					Expect(err).To(BeNil())
+					deleteAffineBusyboxPod(preflightBusyboxPod, defaultTestNs)
+				})
+			})
+
+			Context("Preflight run command, pod affinity test cases with pods not able to schedule on node of a cluster", func() {
+				var (
+					nodeList     *corev1.NodeList
+					testNodeName string
+					node         *corev1.Node
+				)
+
+				BeforeEach(func() {
+					nodeList, err = k8sClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+					Expect(err).To(BeNil())
+					node = &nodeList.Items[0]
+					testNodeName = node.GetName()
+					nodeLabels := node.GetLabels()
+					nodeLabels[preflightNodeLabelKey] = preflightNodeLabelValue
+					node.SetLabels(nodeLabels)
+					node, err = k8sClient.CoreV1().Nodes().Update(ctx, node, metav1.UpdateOptions{})
+					Expect(err).To(BeNil())
+					createAffineBusyboxPod(preflightBusyboxPod, highAffinity, defaultTestNs)
+				})
+
+				It("Should not be able to schedule DNS and source pod on any node of the cluster", func() {
+					yamlFilePath := filepath.Join(testDataDirRelPath, podAffinityInputFile)
+					inputFlags := make(map[string]string)
+					copyMap(flagsMap, inputFlags)
+					inputFlags[configFileFlag] = yamlFilePath
+					cmdOut, err = runPreflightChecks(inputFlags)
+					Expect(err).ToNot(BeNil())
+
+					Expect(cmdOut.Out).To(MatchRegexp("DNS pod - dnsutils-[a-z]{6} hasn't reached into ready state"))
+					Expect(cmdOut.Out).To(ContainSubstring(
+						"Preflight check for DNS resolution failed :: timed out waiting for the condition"))
+
+					Expect(cmdOut.Out).To(MatchRegexp(
+						"Preflight check for volume snapshot and restore failed :: pod source-pod-[a-z]{6} hasn't reached into ready state"))
+				})
+
+				AfterEach(func() {
+					node, err = k8sClient.CoreV1().Nodes().Get(ctx, testNodeName, metav1.GetOptions{})
+					nodeLabels := node.GetLabels()
+					delete(nodeLabels, preflightNodeLabelKey)
+					node.SetLabels(nodeLabels)
+					node, err = k8sClient.CoreV1().Nodes().Update(ctx, node, metav1.UpdateOptions{})
+					Expect(err).To(BeNil())
+					deleteAffineBusyboxPod(preflightBusyboxPod, defaultTestNs)
+				})
+			})
+
+			Context("Preflight run command, taints and tolerations test cases with pods able to schedule on node of a cluster", func() {
+				var (
+					nodeList      *corev1.NodeList
+					taintNodeName string
+					node          *corev1.Node
+				)
+
+				BeforeEach(func() {
+					nodeList, err = k8sClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+					Expect(err).To(BeNil())
+					node = &nodeList.Items[0]
+					taintNodeName = node.GetName()
+					nodeLabels := node.GetLabels()
+					nodeLabels[preflightNodeLabelKey] = preflightNodeLabelValue
+					node.SetLabels(nodeLabels)
+					nodeTaints := node.Spec.Taints
+					nodeTaints = append(nodeTaints, corev1.Taint{
+						Key:    preflightTaintKey,
+						Value:  preflightTaintValue,
+						Effect: corev1.TaintEffectNoSchedule,
+					})
+					node.Spec.Taints = nodeTaints
```

Comment:
do not add taints .. for this CI node, it'll stop scheduling pods of all CI jobs